aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -304,3 +282,108 @@ pub(crate) fn validate_authenticators(
 
     Ok(())
 }
+
+fn verify_zk_sig(
+    jwk: &JWK,
+    zk_sig: &ZeroKnowledgeSig,
+    sig: &KeylessSignature,
+    pk: &KeylessPublicKey,
+    config: &Configuration,
+    training_wheels_pk: Option<&EphemeralPublicKey>,
+    pvk: Option<&PreparedVerifyingKey<Bn254>>,
+) -> Result<(), VMStatus> {
+    match jwk {
+        JWK::RSA(rsa_jwk) => {
+            if zk_sig.exp_horizon_secs > config.max_exp_horizon_secs {
+                // println!("[aptos-vm][groth16] Expiration horizon is too long");
+                return Err(invalid_signature!("The expiration horizon is too long"));
+            }
+
+            // If an `aud` override was set for account recovery purposes, check that it is
+            // in the allow-list on-chain.
+            if zk_sig.override_aud_val.is_some() {
+                config.is_allowed_override_aud(zk_sig.override_aud_val.as_ref().unwrap())?;
+            }
+
+            match &zk_sig.proof {
+                ZKP::Groth16(groth16proof) => {
+                    // let start = std::time::Instant::now();
+                    let public_inputs_hash = get_public_inputs_hash(sig, pk, rsa_jwk, config)
+                        .map_err(|_| {
+                            // println!("[aptos-vm][groth16] PIH computation failed");
+                            invalid_signature!("Could not compute public inputs hash")
+                        })?;
+                    // println!("Public inputs hash time: {:?}", start.elapsed());
```

Comment:
Do we need to have these printlns and other commented out code?

Reply:
no, will delete.